lib/matplotlib/legend.py

```diff
@@ -1145,6 +1142,12 @@ def _find_best_position(self, width, height, renderer, consider=None):
             candidates.append((badness, idx, (l, b)))
 
         _, _, (l, b) = min(candidates)
+
+        if self._loc_used_default and time.perf_counter() - start_time > 5:
```

Comment:
I would go with a threshold of 1 s or 0.5s

The font-manager warning in a one-time start up issue, this is a on-every-draw cost.

Reply:
switched to 1s, which also makes the test faster :p